### DISCUSSION TITLE
Gen 4 X1 Hybrid suddenly stopped working with integration

### DISCUSSION THREAD

**Discussion by M0LTE** [wills106/homeassistant-solax-modbus#262] [Q&A]
My gen 4 SolaX X1 Hybrid + ethernet modbus adapter combo has suddenly stopped working with this integration - my sensors at least are all "Unknown" value.

I have removed and re-added the integration with no success.

Where do I start please?

I have logs:

Logger: custom_components.solax_modbus
Source: /config/custom_components/solax_modbus/__init__.py:369
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 8:18:02 PM (1 occurrences)
Last logged: 8:18:02 PM

SolaX error reading holding registers at device 1 position 0x200
NoneType: None

Logger: custom_components.solax_modbus
Source: custom_components/solax_modbus/const.py:182
Integration: SolaX Inverter Modbus (documentation, issues)
First occurred: 8:18:02 PM (3 occurrences)
Last logged: 8:22:47 PM

Something went wrong reading from modbus
Traceback (most recent call last):
  File "/config/custom_components/solax_modbus/__init__.py", line 320, in read_modbus_data
    res = self.read_modbus_registers_all()
  File "/config/custom_components/solax_modbus/__init__.py", line 397, in read_modbus_registers_all
    self.data[descr.key] = descr.value_function(0, descr, self.data )
  File "/config/custom_components/solax_modbus/const.py", line 182, in value_function_grid_import
    val = datadict["measured_power"]
KeyError: 'measured_power'

**Comment by wills106** [comment]
Can you roll back to 2022.12.6?

I added in some extra BMS info after 2022.12.6 and it seems like not all Inverters like it. A Gen3 is the same but:

> SolaX error reading holding registers at device 1 position 0x206

As they are in different places between Gen3 and Gen4

But you both get the following:
> val = datadict["measured_power"]
KeyError: 'measured_power'

Not sure why at the min? I'll probably remove that whole BMS section as it's pointless unless you want to know your battery serial number.

**Reply by wills106** [reply]
How many batteries do you have connected to your Inverter?

**Reply by M0LTE** [reply]
3 - one slave, two master.

**Comment by infradom** [comment]
@wills106 Why is it reading holding register 0x200 ?  I dont see any reason why it should read that address

**Reply by infradom** [reply]
sorry BMS registers

**Comment by M0LTE** [comment]
About to attempt downgrade

**Reply by M0LTE** [reply]
@wills106 Downgrade successful and I've got my sensors back. Thanks.

**Reply by infradom** [reply]
The BMS entities worked on my Gen4  after I enabled a couple of them
Tom, did you manually enable BMS options when it went wrong? They are disabled by default if I am right ..

**Comment by wills106** [comment]
Another report #263 (Third now)

Going to remove releases after 2022.12.6 and then remove this BMS stuff.

**Comment by M0LTE** [comment]
I am talking about having lost all of my basic sensors like battery state
of charge, current power usage etc.

Didn’t manually enable any disabled entities.

I must have not spotted it was broken after doing my recent upgrade to
latest.

**Reply by wills106** [reply]
As you didn't enable the BMS sensors it shouldn't be reading them anyway.
We will look into that.

Also it's possible these registers are missing on certain Inverters. I don't know if this is a firmware issue or a hardware revision issue.
I have had this in the past though where Inverters are missing registers.
